app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -691,6 +693,9 @@ open class BrowserActivity : DuckDuckGoActivity() {
             pixel.fire(FIRE_DIALOG_CANCEL)
             currentTab?.onFireDialogVisibilityChanged(isVisible = false)
         }
+        dialog.clearStarted = {
+            isDataClearingInProgress = true
```

Comment:
Can you move this code at line https://github.com/duckduckgo/Android/pull/5903/files#diff-53f6cb0433b1d43092e96b2b95aa39158ce4fb1536a91a17540da459a5835c0eR689?

I.e instead of:
```
dialog.clearStarted = {
            isDataClearingInProgress = true
}
```

Modify the above code to:
```
dialog.clearStarted = {
            isDataClearingInProgress = true
            removeObservers()
}
```

Reply:
Oops, I missed that this callback was already used. Thanks!